Add contributing guidelines to repos (#75)
holochain/hc-github-config: files/CONTRIBUTING.md

Author: I'm not sure if we need all these sections as I think the guide is mainly for us but I thought it might be nice to have more clear instructions for any external people.

# Contributing to {{REPO_NAME}}

Thank you for your interest in contributing to {{REPO_NAME}}!

## Getting Started

1. Fork the repository

> 2. Create a feature branch from `main`

Author: Should this be `develop` or "the default branch"

Reviewer: The default branch, the one that shows up on the homepage of the repository above the files in the branch selection drop down.

Reviewer 2: Except for the holochain repo, which isn't managed by Pulumi, the default branch is `main` in nearly all repos. Could say "the default branch (usually `main`)".

Author: Done in 5a64278017f93e286df8b0fb5015ba108ae66a44

3. Make your changes
4. Submit a pull request

## Commits

The commits are used to generate the changelog upon a release, therefore keep
them clean. To help with keeping them clean, follow these principles:

- Use [conventional commits](https://www.conventionalcommits.org/) for commit
  messages. You can also use markdown, especially in the body
- Keep commits atomic, meaning that each commit should build and the unit tests
  should pass. If a new feature is added or a bug is fixed then fix the tests
  in the same commit

> - Always rebase, **NEVER** merge

Reviewer 2: ```suggestion
- In case of changes on the origin branch, always rebase, **NEVER** merge
```

- All changes require a pull request and at least one approving review
- Add a short, clear, and hand-written description of the change proposed by
  the PR. AI tooling may add an in-depth summary below your description so keep
  it high-level but make sure to add one
- PRs must pass CI checks before merging
- Keep PRs focused — one logical change per PR
- Add comments to your own PR before requesting review to explain the approach
  or to add questions for the reviewers
- Ensure all review threads are resolved before merging
- Use `fixup!` commits to address review comments and make additional changes
  once the PR is open, this allows the reviewers to see what has changed
  between reviews
- When addressing a review comment, link to the commit that addressed the
  comment, even if it is a `fixup!` commit that will be squashed later
- After an approval, rebase on the base branch. If this is done via the GitHub
  UI then no re-approval will be needed. However, if this is done locally then
  a re-approval is required
- Squash all `fixup!` commits and clean-up the commit history before merging
  into the base branch. You can do this at the same time as rebasing with

>   `git rebase --autosquash <base_branch>`

Reviewer: This doesn't work for me, I always use `-i` at the same time. Maybe I'm doing something wrong there :)

Reviewer 2: It works for me on macos and Linux. You must be doing something wrong! 😛

Author: What version of Git are you on @ThetaSinner? It seems that the `--interactive` requirement was removed in [Git v2.44.0](https://gitlab.com/git-scm/git/-/blob/HEAD/Documentation/RelNotes/2.44.0.adoc). Released in January 2025, maybe you should update but maybe I should also add this note in the doc?

Author: Added a note in 103a55e97d78df2e611f1e2aab6601c001cfdbbe

Reviewer: I blame Linux package management :)

```
git is already the newest version (1:2.43.0-1ubuntu7.3).
```

- Keep PRs focused — one logical change per PR.
- Add comments to your own PR before requesting review to explain the approach
  or to add questions for the reviewers.
- Ensure all review threads are resolved before merging.
- Use `fixup!` commits to address review comments and make additional changes
  once the PR is open, as this allows the reviewers to see what has changed
  between reviews.
- When addressing a review comment, link to the commit that addressed the
  comment, even if it is a `fixup!` commit that will be squashed later.
- After an approval, rebase on the base branch. If this is done via the GitHub
  UI, no re-approval will be needed. However, if this is done locally, then a
  re-approval is required.
- Squash all `fixup!` commits and clean up the commit history before merging
  into the base branch. You can do this at the same time as rebasing with
  `git rebase --autosquash <base_branch>`. For older versions of git, you may
  need to use `git rebase --autosquash --interactive <base_branch>`, which will
  perform an interactive rebase with all of the `fixup!` commits rearranged and
  marked as fixup, you can then accept this interactive rebase.

> - We reserve the right to reject PRs that are purely cosmetic and appear to

Reviewer 3: ```suggestion
- We will reject PRs that are purely cosmetic and appear to
```

"reserve the right" is a bit strange wording, like a legal contract. I would just say the intention that we "will", or "will usually", or "may"

Author: Done in f1e1ac1445f4

  have been automated with tooling, such as correcting spelling and grammatical
  mistakes. If you believe that such a change is critical in making the
  documentation or code more readable, or if the change corrects a real logical
  mistake in the text, then submit such a PR with a description explaining the
  critical correction.
- All AI-generated PRs must have been self-reviewed. If a PR is judged to be
  AI-generated, not checked by the author, and needs a lot of work to be
  consistent with the existing codebase and/or it does not solve the problem it

>   claims to, then we reserve the right to close the PR without comment.

Reviewer 3: ```suggestion
  claims to, then we will close the PR without comment.
```

same here

Author: Done in f1e1ac1445f4
